lurk-derive/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+//! Derive macros for Lurk
+//! This crate contains derive macros to manage trait dispatch in Lurk.
+//!
+//! - The `Coproc` macro adds dispatching `Coprocessor` and `Cocircuit` implementaitons to enums whose variants all
```

Comment:
```suggestion
//! - The `Coproc` macro adds dispatching `Coprocessor` and `Cocircuit` implementations to enums whose variants all
```

Reply:
> So, concept ACK, and the code looks fine, but we already have a proc-macro crate (lurk_macro). Why not rename it to lurk-derive and have its contents alongside those that were there (i.e. merge this PR's crate and the existing one)? Is there a compelling reason to have two?

No reason, will change. I just got excited about writing my first proc macro, which erased my memory of having written one before.

Reply:
Actually yes, there is a reason. The `lurk_macro` uses the `lurk` crate, so it cannot be used *in* `lurk`.

Reply:
@porcuquine this dependency is unused.